### PR TITLE
Optimize chat manager logic

### PR DESCRIPTION
**Pull request by P3il4**
This PR optimizes the way the chat manager (manage chat files button) works by moving the search logic into the backend. This way it's no longer necessary to do hundreds of HTTP requests to get every chat file. As the server returns only the chat previews instead of the entire chat data, network load is also reduced.
For me it reduced chat manager load time from 20-30s (over a minute with a bad connection) to 1-2s with ~120 chat files on a remotely hosted ST instance.

Testing changes:
- Open chat manager for a character
- Open chat manager for a group
- Search for a specific chat in chat manager
Preferably have 100-200 large chats and access ST over a remote connection to test performance gain.

edit: fixed a bug with chat deletion

## Checklist:

- [x] I have read the [Contributing guidelines](https://github.com/SillyTavern/SillyTavern/blob/release/CONTRIBUTING.md).

### CODE REVIEW

**Comment by P3il4** [comment]
sorry i'm bad at git stuff
fixed issue if character wasn't named the same as the .png and fixed converting chats to groups
also tested renaming and exporting chats. everything should work now

File: public/script.js [hunk 1]
```diff
@@ -10421,6 +10384,11 @@ jQuery(async function () {
         $('#load_select_chat_div').css('display', 'block');
     });
 
+    // not sure what that hourglass was for
```
Comment (Cohee1207) [inline public/script.js]: It's an [ancient TAI artifact](https://github.com/TavernAI/TavernAI/blob/main/public/index.html#L460). Could be removed safely.

**Comment by Cohee1207** [comment]
I fixed some issues:
- Missing selected highlight
- Incorrect solo chat counters
- Sorting by name instead of timestamp

Will continue testing for now.

**Comment by Cohee1207** [comment]
> sorry i'm bad at git stuff

You don't need to squash and force push every commit. Just do them one by one.

**Review by Cohee1207 — APPROVED** [review]
I don't think I have anything more to add. Good job.

**Pull request merged**